.github/workflows/industrial_ci.yml

```
@@ -0,0 +1,26 @@
+name: industrial_ci
+
+on:
+  push:
+    paths-ignore:
+    - '**.md'
+  pull_request:
+    paths-ignore:
+    - '**.md'
+  schedule:
+    - cron: "0 2 * * 0" # Weekly on Sundays at 02:00
+
+jobs:
+  industrial_ci:
+    strategy:
+      matrix:
+        distro: [ kinetic, melodic ]
+    env:
+      ROS_REPO: ros
+      ROSDEP_SKIP_KEYS: roboticsgroup_gazebo_plugins
```

```suggestion
    continue-on-error: ${{ matrix.experimental }}
    strategy:
      matrix:
        env:
          - { ROS_DISTRO: kinetic, ROS_REPO: main, UPSTREAM_WORKSPACE: .rosinstall }
          - { ROS_DISTRO: melodic, ROS_REPO: main, UPSTREAM_WORKSPACE: .rosinstall }
        experimental: [false]
        include:
          - env: { ROS_DISTRO: kinetic, ROS_REPO: testing, UPSTREAM_WORKSPACE: .rosinstall }
            experimental: true
          - env: { ROS_DISTRO: melodic, ROS_REPO: testing, UPSTREAM_WORKSPACE: .rosinstall }
            experimental: true
```
https://github.com/rt-net/raspicat_slam/blob/master/.kinetic_depends.rosinstall のように`.rosinstall`を作り`UPSTREAM_WORKSPACE`に指定すると`python-vcstool`でパッケージを追加できるようです。
https://github.com/ros-industrial/industrial_ci/blob/master/doc/index.rst#optional-environment-variables

また、https://github.com/rt-net/raspimouse_sim/pull/39/files#diff-67a83e58eae82bfd5f88e51393c29834R17-R28
上記の変更のように`ROS_REPO`は`main`（現在のaptリリース）と`testing`（次のaptリリース）を加えてみてはどうでしょうか。

---

なるほど！ファイルを作るんですね！

ROS_REPOの件も了解です。やってみます。